diktat-maven-plugin/src/main/kotlin/org/cqfn/diktat/plugin/maven/DiktatMojo.kt

```diff
@@ -0,0 +1,44 @@
+/**
+ * MOJOs for goals of diktat plugin
+ */
+
+package org.cqfn.diktat.plugin.maven
+
+import com.pinterest.ktlint.core.KtLint
+import java.io.File
+import org.apache.maven.plugins.annotations.Mojo
+import org.cqfn.diktat.ruleset.rules.DiktatRuleSetProvider
+
+/**
+ * Main [Mojo] that call [DiktatRuleSetProvider]'s rules on [inputs] files
+ */
+@Mojo(name = "check")
+class DiktatCheckMojo : DiktatBaseMojo() {
+    /**
+     * @param params instance of [KtLint.Params] used in analysis
+     */
+    override fun runAction(params: KtLint.Params) {
+        KtLint.lint(params)
+    }
+}
+
+/**
+ * Main [Mojo] that call [DiktatRuleSetProvider]'s rules on [inputs] files
+ * and fixes discovered errors
+ */
+@Mojo(name = "fix")
+class DiktatFixMojo : DiktatBaseMojo() {
+    /**
+     * @param params instance of [KtLint.Params] used in analysis
+     */
+    override fun runAction(params: KtLint.Params) {
+        val fileName = params.fileName
+        val filePath = params.userData["file_path"] ?: error("File path should be provided")
+        val fileContent = File(filePath).readText(charset("UTF-8"))
+        val formattedText = KtLint.format(params)
+        if (fileContent !== formattedText) {
+            log.info("Original and formatted content differ, writing to $fileName...")
+            File(filePath).writeText(formattedText, charset("UTF-8"))
```

Comment:
are you sure we really need to write anything to a file? 
What's with the original writer from ktlint?

WHat you gonna do with \n. I guess you need to convert newlines

Reply:
Substitution of `\n` is done inside `KtLint.lint/format`.

Calling any actions on input files is done on a higher level, inside `Main.kt` in ktlint. If we want to customize behavior, e.g. have a way to provide path to diktat config file, we should be able to change that logic, that's why I re-implemented part of it.

Reply:
no, no need to customize it

Reply:
I mean initial logic, including rule set construction. All file processing related stuff is done inside `KtLint.lint` and I don't override anything there.